pinot-spi/src/main/java/org/apache/pinot/spi/config/table/TimestampIndexGranularity.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.config.table;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.pinot.spi.data.DateTimeFieldSpec;
+import org.apache.pinot.spi.data.DimensionFieldSpec;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.apache.pinot.spi.data.MetricFieldSpec;
+
+
+/**
+ * TimestampIndexGranularity is the enum of different time granularities from MILLIS to YEAR.
+ */
+public enum TimestampIndexGranularity {
+  MILLISECOND, SECOND, MINUTE, HOUR, DAY, WEEK, MONTH, QUARTER, YEAR;
+
+  private static Set<String> _validValues =
+      Arrays.stream(TimestampIndexGranularity.values()).map(v -> v.toString()).collect(Collectors.toSet());
+
+  public static String getColumnNameWithGranularity(String column, TimestampIndexGranularity granularity) {
+    return "$" + column + "$" + granularity;
+  }
+
+  public static String extractColumnNameFromColumnWithGranularity(String columnWithGranularity) {
+    return StringUtils.split(columnWithGranularity, "$")[0];
```

Comment:
If we care about performance here let's not allocate an array just to take the first element ;)

```java
  public static String getFirstValue(String value) {
    // use a string separator because the char overload has only been intrinsified since JDK16
    int index = value.indexOf("$");
    return index < 0 ? value : value.substring(0, index);
  }
```

Reply:
I will let you both decide :p

Reply:
Let's wait and see if it shows up in profiles

Reply:
I'd suggest only split once and keep the array since we need both parts.
FYI we need to use `value.indexOf("$", 1)` because the first character is `'$'`

Reply:
Actually, after the recent refactoring, this method along with `extractGranularity` method is not used anymore.
I will just delete it.